internal/saucecloud/cloud.go

```diff
@@ -934,3 +966,115 @@ func (r *CloudRunner) getHistory(launchOrder config.LaunchOrder) (insights.JobHi
 	}
 	return r.InsightsService.GetHistory(context.Background(), user, launchOrder)
 }
+
+func (r *CloudRunner) reportSuiteToInsights(res result) []insights.TestRun {
+	// Skip reporting if job is not completed
+	if !job.Done(res.job.Status) || res.skipped || res.job.ID == "" {
+		return []insights.TestRun{}
+	}
+
+	assets, err := r.JobService.GetJobAssetFileNames(context.Background(), res.job.ID, res.job.IsRDC)
+	if err != nil {
+		// TODO: Update message
+		log.Warn().Err(err).Msg(msg.InsightsReportError)
+	}
+
+	var testRuns []insights.TestRun
+
+	if arrayContains(assets, saucereport.SauceReportFileName) {
+		report, err := r.loadSauceTestReport(res.job.ID, res.job.IsRDC)
+		if err != nil {
+			log.Warn().Err(err).Str("action", "parsingJSON").Msg(msg.InsightsReportError)
```

Comment:
just curious about the error handling. If there is a loading error for the test report, it still needs to generate an empty `testRuns`?

Reply:
It was not expected, removing that. 
Also removing the return value that make no sense now.